backend/dataall/modules/s3_datasets_shares/services/s3_share_managed_policy_service.py

```diff
@@ -128,6 +130,23 @@ def check_resource_in_policy_statement(target_resources: list, existing_policy_s
                 return False
         return True
 
+    @staticmethod
+    def check_s3_actions(existing_policy_statement: dict) -> (bool, str, str):
+        """
+        Checks if all required s3 actions are allowed in the existing policy and there is no unallowed actions
+        :param existing_policy_statement:
+        :return: bool, allowed missing actions string, not allowed actions string
+        """
+        statement_actions = set(existing_policy_statement['Action'])
+        allowed_actions = set(S3_ALLOWED_ACTIONS)
+        missing_actions = allowed_actions - statement_actions
+        extra_actions = statement_actions - allowed_actions
+        return (
+            not (missing_actions or extra_actions),
```

Comment:
nit: probably not required, since you return missing and extra actions the caller can decide using the same code

Reply:
It's just shorter. And also, may be one day we will decide, that it's not an error, but just a warning

Reply:
Exactly, the caller might decide what's right for them i.e maybe access_points were fine with missing permissions and s3 isn't. Anyway it's not big deal.